web3.js/src/transaction/versioned.ts

```diff
@@ -106,4 +107,19 @@ export class VersionedTransaction {
       this.signatures[signerIndex] = sign(messageData, signer.secretKey);
     }
   }
+
+  addSignature(publicKey: PublicKey, signature: Uint8Array) {
+    const signerPubkeys = this.message.staticAccountKeys.slice(
```

Comment:
The legacy implementation has a check here for signature length. Maybe we should add this too.

```typescript
assert(
  signature.byteLength === 64,
  'Signature must be 64 bytes long',
);
```